using-timescaledb/custom-jobs.md

```diff
@@ -0,0 +1,221 @@
+# Custom Jobs
+
+Custom jobs allow you to run functions and procedures implemented in a
+language of your choosing on a schedule within TimescaleDB. This allows
+automatic periodic tasks that are not covered by existing policies and
+even enhancing existing policies with additional functionality.
+
+### Creating a custom job procedure [](create-procedure)
+
+The signature for custom jobs is (job_id INT, config JSONB).
+The content of the config JSONB is completely up to the job and may
+also be NULL if no variables are required.
+
+Template for a custom job procedure.
+
+```sql
+CREATE OR REPLACE PROCEDURE custom_job(job_id int, config jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+	RAISE NOTICE 'Executing job % with config %', job_id, config;
+END
+$$;
+```
+
+### Registering your custom job [](register)
+
+When registering the custom job `job_id` and `config` are stored in the
+TimescaleDB catalog. The `config` JSONB can be modified with `alter_job`.
+`job_id` and `config` will be passed as arguments when the procedure is
+executed as background process or when run with `run_job`.
+
+```sql
+SELECT add_job('custom_job','1h');
+```
+
+Register the created job with the automation framework. `add_job` returns the job_id
+which can be used to execute the job manually with `run_job`.
+
+```sql
+CALL run_job(1000);
+```
+
+Execute job with id 1000 in current session.
+
+### Testing/Debugging jobs [](testing)
+
+Any background worker job can be run in foreground when executed with `run_job`. This can
+be useful to debug problems when combined with increased log level.
+
+### Altering / Dropping custom jobs [](alter-delete)
+
+You can alter the config or scheduling parameters with `alter_job`.
+
+```sql
+SELECT alter_job(1000, config => '{"hypertable":"metrics"}');
+```
+
+Replaces the config for job with id 1000 with the specified JSON.
+
+```sql
+SELECT alter_job(1000, scheduled => false);
+```
+
+Disables automatic scheduling of the job with id 1000. The job can still be run manually
+with `run_job`.
+
+```sql
+SELECT alter_job(1000, scheduled => true);
+```
+
+Reenables automatic scheduling of the job with id 1000.
+
+```sql
+SELECT delete_job(1000);
+```
+
+Deletes the job with id 1000 from the automation framework.
+
+## Example custom jobs [](examples)
+
+### Generic Retention Policy [](generic-retention-policy)
+
+Retention policy for multiple hypertables. Additional filters can be applied
+by adding a WHERE clause to the query.
+
+```sql
+CREATE OR REPLACE PROCEDURE generic_retention (job_id int, config jsonb)
+LANGUAGE PLPGSQL
+AS $$
+DECLARE
+  lag interval;
```

Comment:
I know there was some discussion about what the "generic" term we should use for this concept of `lag`.  Have we finalized on `lag` as the name throughout?

Reply:
all the policies have the _after suffix for this type of arguments, e.g. compress_after